pennylane/transforms/core/transform_program.py

```diff
@@ -341,6 +343,25 @@ def has_classical_cotransform(self) -> bool:
         """
         return any(t.classical_cotransform is not None for t in self)
 
+    def prune_dynamic_transform(self):
+        """Ensure a single ``dynamic_one_shot`` transform is applied."""
+        trans_type = np.zeros(len(self._transform_program), dtype=np.int32)
+        for i, t in enumerate(self._transform_program):
+            if "dynamic_one_shot" in str(t):
+                trans_type[i] = 1
+            if "mid_circuit_measurements" in str(t):
+                trans_type[i] = 2
+        if sum(trans_type) < 2:
+            return
+        keep = 2 if 2 in trans_type else 1
+        found = False
+        for i, type in enumerate(reversed(trans_type)):
+            if not found and type == keep:
+                found = True
+                continue
+            if type in [1, 2]:
```

Comment:
nitpick but i prefer not to override builtins
```suggestion
        for i, ttype in enumerate(reversed(trans_type)):
            if not found and ttype == keep:
                found = True
                continue
            if ttype in [1, 2]:
```

Reply:
100% agree, it just slipped my mind.